### ISSUE TITLE
Mono update/upgrade failure

### ISSUE THREAD

**Issue by PedEchsing** [mono/mono#21461]
relative newbie here... just doing what I've learned from OS X (homebrew).
mono-devel_6.12.0.122-0xamarin1+ubuntu2004b1_all.deb will NOT overwrite/update 'usr/bin/disco'.
Running Ubuntu Studio (wanted 'some' support for ipod-touch).
Was trying to install iTunes under Wine. Had read/was told it worked and was the "best" way to
connect Touch for music maint. and control. Neither Wine or iTunes worked. Probably broke Mono.
Can't force re-install, can't re-install, can't apt, can't update with the built-in 'Discover'.

I probably(!) won't see any responses to the bug-report, I'm usually in Windows or on my Mac in
High Sierra and the github account is setup under a different login. That doesn't mean you shouldn't try,
just don't expect any responses.
Just wanted to get the report listed so it can be fixed in future updates.

**Comment by ZeHipo** [comment]
having the same problem trying to update mono-devel on ubuntu studio 22.04. 
there seems to be a naming conflict with discodos 1.0~rc2-2:
 https://github.com/JOJ0/discodos

**Comment by ZeHipo** [comment]
for anyone encountering this issue:
you can purge discodos like proposed in this thread:
https://bugs.launchpad.net/ubuntu/+source/mono/+bug/1928414

unfortunately discodos and mono both use the 'disco' command, probably making it impossible to install both packages on the same system at the same time as of right now.

**Comment by PedEchsing** [comment]
OK GOT IT!!
First I tried ZeHippo's link
sudo dpkg --remove --force-remove-reinstreq mono-devel
Then tried using Discover to update... no change.
Then I tried
sudo dpkg -l | grep '^rc' | awk '{print $2}' | xargs dpkg --purge discodos
Had to use sudo su to run it.
Ran it as is, ran it under 'sudo su', than ran it under standard 'sudo' again.
After the 3rd run, Ubuntu reported 'not installed' so I then rebooted.
Re-ran Discover, ran the update, fully updated this time (no errors).

I don't know if it was the second command or the reboot but its fixed!
If an Administrator can close this?... (will try 'Close with comment')

THANK YOU TO EVERYONE'S HELP.
